planner/core/exhaust_physical_plans.go

```diff
@@ -363,7 +363,7 @@ func (p *LogicalJoin) constructIndexJoin(
 	if path != nil {
 		join.IdxColLens = path.idxColLens
 	}
-	join.SetSchema(p.schema)
+	join.SetSchema(p.schema.Clone())
```

Comment:
Why need clone them?

Reply:
If using `p.schema`, they will share a schema. But the schema shouldn't be shared. For example, the schames of indexScan and tableScan may be different.(indexScan with virtual column don't expand its dependent columns)

Reply:
Sorry, the `Clone()` here is not necessary.